lib/geminabox/external_dependency.rb

```diff
@@ -0,0 +1,28 @@
+require 'json'
+
+module Geminabox
+  module ExternalDependency
+
+    class << self
+
+      def for(*gems)
+        source_list = []
+        Sources.external_sources.each do |source|
+          url = [ external_uri(source), '?gems=', gems.map(&:to_s).join(',') ].join
+          body = Geminabox.http_adapter.get_content(url)
+          source_list << JSON.parse(body)
+        end
+        source_list.flatten.uniq
+      rescue Exception => e
+        return [] if Geminabox.allow_remote_failure
```

Comment:
Looks like the logic is not quite right here. Currently if any of the remotes fail the entire for method will return an empty array. In reality I would like to see fail over for each source provided allow_remote_failure = true.